[BUG]: Compilation Error with nginx 1.26.3 and nginx-datadog v1.6.0 - Missing <stdexcept> Include

### Module Version(s)

v1.6.0

### Bug Report

**Description**:
When compiling the DataDog nginx module against nginx version 1.26.3 using nginx-datadog v1.6.0, the build fails with a compilation error related to missing std::out_of_range declaration.

**Environment**:

nginx version: 1.26.3

nginx-datadog version: 1.6.0

Compilation command: NGINX_VERSION=${NGINX_VERSION} make build

```
/tmp/nginx-datadog/src/tracing/directives.cpp: In function 'char* datadog::nginx::set_datadog_sample_rate(ngx_conf_t*, ngx_command_t*, void*)':
/tmp/nginx-datadog/src/tracing/directives.cpp:104:18: error: 'out_of_range' is not a member of 'std'
  104 |       throw std::out_of_range("");  // error message is in the `catch` handler
      |                  ^~~~~~~~~~~~
/tmp/nginx-datadog/src/tracing/directives.cpp:7:1: note: 'std::out_of_range' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
    6 | #include "ngx_http_datadog_module.h"
  +++ |+#include <stdexcept>
    7 | 
```
**Workaround**:
Downgrading to nginx-datadog v1.5.0 resolves the compilation issue.

**Suggested Fix**:
The error suggests that <stdexcept> header needs to be included in src/tracing/directives.cpp. This is a straightforward fix that should be added to the source code.

**Additional Information**:
I'm happy to provide:

Full build logs

System information

Test the fix once available

Any other details that would help resolve this issue

Let me know what other information would be helpful for troubleshooting this issue.

### Reproduction Code

```
git clone --branch v1.6.0 https://github.com/DataDog/nginx-datadog.git
cd nginx-datadog/ || exit 1

NGINX_VERSION=$(get_nginx_version)

echo "Nginx version detected: ${NGINX_VERSION}"

NGINX_VERSION=${NGINX_VERSION} make build || exit 2
```

### Error Logs

_No response_

### Operating System

_No response_

@dmehala  thank for the response

### Additional Information

Here are a few more environment details that might help with debugging:

- **Compiler version:**
  ```bash
  gcc --version
  gcc (Alpine 13.2.1_git20240309) 13.2.1 20240309

- **Make version:**
  ```bash
  make --version
  GNU Make 4.4.1

- **CPU architecture:**
  ```bash
  x86_64
- **Nginx version:**
  ```bash
  1.26.3



The reproduction code mentioned earlier is part of a script executed during the Docker build process, like this:
 ```bash
    RUN chmod +x datadog-module.sh && ./datadog-module.sh
```

- **What I Tried:**

I attempted your approach using `build-musl` with my current Nginx version (1.26.3) and without enabling WAF, but encountered the following error:
```bash
  command not found
  make: *** [Makefile:105: build-musl] Error 127 
```
***Context:***

We've been using this same script for a year without any issues. However, starting around March 11, the build began failing. After reverting back to Datadog version 1.50, the script started working again.


***Note:***

I'm not deeply familiar with compilers or Nginx module internals, so I apologize if I'm missing or miscommunicating any important technical details. I truly appreciate your patience and guidance as we work through this.

--- a/src/tracing/directives.cpp
+++ b/src/tracing/directives.cpp
@@ -1,6 +1,7 @@
 #include "tracing/directives.h"
 
 #include <cassert>
+#include <stdexcept>
 
 #include "common/variable.h"
 #include "ngx_http_datadog_module.h"

